erpnext/controllers/stock_controller.py

```diff
@@ -250,6 +250,7 @@ def get_sl_entries(self, d, args):
 			"voucher_type": self.doctype,
 			"voucher_no": self.name,
 			"voucher_detail_no": d.name,
+			"selling_rate": d.get('base_rate', 0) if not self.doctype.startswith("Purchase") else 0,
```

Comment:
It will be better to have explicit doctypes from where we can get selling rate.
In this case, `self.doctype in ('Delivery Note', 'Sales Invoice')`

Reply:
change request is done